src/Aspire.Dashboard/Components/Controls/MetricTable.razor.cs

```diff
@@ -0,0 +1,401 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Aspire.Dashboard.Model;
+using Aspire.Dashboard.Otlp.Model;
+using Aspire.Dashboard.Otlp.Model.MetricValues;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+using CollectionExtensions = Aspire.Dashboard.Extensions.CollectionExtensions;
+
+namespace Aspire.Dashboard.Components.Controls;
+
+public partial class MetricTable : ComponentBase
+{
+    private readonly List<MetricView> _metrics = [];
+    private bool _onlyShowValueChanges;
+    private bool _anyDimensionsShown;
+    private IJSObjectReference? _jsModule;
+
+    private OtlpInstrument? _instrument;
+    private bool _showCount;
+
+    private IQueryable<MetricView> _metricsView => _metrics.AsEnumerable().Reverse().AsQueryable();
+
+    [Inject] public required IJSRuntime JS { get; init; }
+
+    [Parameter, EditorRequired] public required InstrumentViewModel InstrumentViewModel { get; set; }
+
+    [Parameter, EditorRequired] public required TimeSpan Duration { get; set; }
+
+    protected override void OnInitialized()
+    {
+        InstrumentViewModel.DataUpdateSubscriptions.Add(OnInstrumentDataUpdate);
+    }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            _jsModule = await JS.InvokeAsync<IJSObjectReference>("import", "/Components/Controls/MetricTable.razor.js");
+        }
+    }
+
+    internal static void UpdateMetrics(
+        List<DimensionScope>? matchedDimensions,
+        List<MetricView> currentMetrics,
+        bool isHistogramInstrument,
+        bool showCount,
+        bool onlyShowValueChanges,
+        out List<MetricView> oldMetrics,
+        out List<int> addedIndices,
+        out bool anyDimensionsShown)
+    {
+        oldMetrics = [.. currentMetrics];
+
+        anyDimensionsShown = false;
+        var shouldShowHistogram = isHistogramInstrument && !showCount;
+
+        if (matchedDimensions is not null)
+        {
+            var newMetrics = new List<MetricView>();
+            anyDimensionsShown = matchedDimensions.Any(dimension => !dimension.Name.Equals(DimensionScope.NoDimensions));
+
+            if (!shouldShowHistogram)
+            {
+                var valuesWithDimensionsByEndDate = matchedDimensions
+                    .SelectMany(dimension => dimension.Values.Select(value => (Value: value, Dimension: dimension)))
+                    .Where(kvp => DateTime.UtcNow.Subtract(kvp.Value.End).CompareTo(TimeSpan.FromSeconds(1)) > 0)
```

Comment:
Is this the 1 second delay? Could you add a comment explaining why to do it.

nit: DateTime supports < and > operators. Those are usually used rather than CompareTo